### PR TITLE
py-numpy: modify depends cython and fftw-3

### PR DESCRIPTION
**Pull request by holymonson**
* move cython to depends_build
* drop obsolete depends fftw-3

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 12.7.6 21H1320 x86_64
Command Line Tools 14.2.0.0.1.1668646533

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by holymonson** [comment]
Did we really need the `gfortran` variant? It seems numpy doesn't require fortran at all.

File: python/py-numpy/Portfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ PortGroup           compiler_blacklist_versions 1.0
 
 name                py-numpy
 version             1.26.4
-revision            2
+revision            3
```
Comment (barracuda156) [inline python/py-numpy/Portfile:10]: Does installed files actually differ? If not (and if the port foes not really use `fftw-3`, nothing links to it anyway), no revbump is due.

**Comment by reneeotten** [comment]
@holymonson thanks I agree with the `fftw-3` dependency that seems to not have been used for a long time by NumPy anymore - some leftover from back in the days. 

Regarding the `cython` dependency I am not completely sure, it does get recorded in `np.show_config()`, so I guess that's why it's kept as a library dependency. But you could be right that it's not actually used at runtime anymore. It is also used for some of the tests, so at least adding it as well to `depends_test` seems like a good idea.

Can you please make that change and then we can merge. It does need the `revision` increase as it is changing `depends_lib` dependencies.

File: python/py-numpy/Portfile (hunk 1 above)
Comment (reneeotten) [reply python/py-numpy/Portfile:10]: It does need the `revision` increase as it changes `depends_lib`

Comment (barracuda156) [reply python/py-numpy/Portfile:10]: Why, if installation is identical?

Comment (reneeotten) [reply python/py-numpy/Portfile:10]: because the dependency is listed in the registry

**Comment by holymonson** [comment]
@reneeotten It was in `depends_test` originally (line 164) and `revision` bumped.

**Comment by jmroot** [comment]
> Regarding the `cython` dependency I am not completely sure, it does get recorded in `np.show_config()`, so I guess that's why it's kept as a library dependency. But you could be right that it's not actually used at runtime anymore. It is also used for some of the tests, so at least adding it as well to `depends_test` seems like a good idea.

I believe the idea may have been that users of `numpy.distutils` could assume that cython was available. Declaring the dependency in the ports that actually use this may be better in some ways, but it might result in a slightly less "batteries included" experience for people building things outside of MacPorts against a MacPorts-provided numpy, and it may also mean that some number of ports that use numpy and cython now need to add an explicit dependency on the latter to avoid breaking.

**Comment by jmroot** [comment]
> Did we really need the `gfortran` variant? It seems numpy doesn't require fortran at all.

https://numpy.org/doc/1.26/user/building.html
> While a FORTRAN 77 compiler is not necessary for building NumPy, it is needed to run the numpy.f2py tests.

**Comment by holymonson** [comment]
AFAIK, on none of other packaging systems (conda, homebrew, archlinux etc.) `cython` is required as a run depends to `numpy`.
For those users realy need `numpy.distutils` to detect `cython`, they will usually require `cython` to precompile their codes. IMHO, it will be better for them to explicitly install `cython` as a dependency.

**Pull request merged**